core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -320,6 +320,31 @@ struct math_function_name;
   };                                                                  \
   constexpr char math_function_name<MathUnaryFunction_##FUNC>::name[]
 
+#define DEFINE_UNARY_FUNCTION_EVAL_CUSTOM(FUNC, ULP_FACTOR, REF_FUNC) \
+  struct MathUnaryFunction_##FUNC {                                   \
+    template <typename T>                                             \
+    static KOKKOS_FUNCTION auto eval(T x) {                           \
+      static_assert(                                                  \
+          std::is_same<decltype(Kokkos::FUNC((T)0)),                  \
+                       math_unary_function_return_type_t<T>>::value); \
+      return Kokkos::FUNC(x);                                         \
+    }                                                                 \
+    template <typename T>                                             \
+    static auto eval_std(T x) {                                       \
+      static_assert(                                                  \
+          std::is_same<decltype(REF_FUNC),                            \
+                       math_unary_function_return_type_t<T>>::value); \
+      return REF_FUNC;                                                \
```

Comment:
passing `decltype(std::sqrt(x))(1) / std::sqrt(x)` below which is a type, not a value

Reply:
This can be evaluated to a value. ?

Reply:
Sorry clearly did not read carefully and thought the decltype encompassed the entire expression.
Which led me to think that the code could not possibly compile...